content/16_TestingRx.md

```diff
@@ -1,846 +1,722 @@
+}
+```
+
+Output:
+
+```
+Time is 50000000 ticks
+Received 5 notifications
+OnNext(0) @ 10000001
+OnNext(1) @ 20000001
+OnNext(2) @ 30000001
+OnNext(3) @ 40000001
+OnCompleted() @ 40000001
+```
+
+Note that our output is exactly the same as the previous example with `Observable.Interval`.
+
+### CreateHotObservable
+
+We can also create hot test observable sequences using the `CreateHotObservable` method. It has the same parameters and return value as `CreateColdObservable`; the difference is that the virtual time specified for each message is now relative to when the observable was created, not when it is subscribed to as per the `CreateColdObservable` method.
+
+This example is just that last "cold" sample, but creating a Hot observable instead.
+
+```csharp
+var scheduler = new TestScheduler();
+var source = scheduler.CreateHotObservable(
+    new Recorded<Notification<long>>(10000000, Notification.CreateOnNext(0L)),
+// ...    
+```
+
+Output:
+
+```
+Time is 50000000 ticks
+Received 5 notifications
+OnNext(0) @ 10000000
+OnNext(1) @ 20000000
+OnNext(2) @ 30000000
+OnNext(3) @ 40000000
+OnCompleted() @ 40000000
+```
+
+Note that the output is almost the same. Scheduling of the creation and subscription do not affect the Hot Observable, therefore the notifications happen 1 tick earlier than their Cold counterparts.
+
+We can see the major difference a Hot Observable bears by changing the virtual create time and virtual subscribe time to be different values. With a Cold Observable, the virtual create time has no real impact, as subscription is what initiates any action. This means we can not miss any early message on a Cold Observable. For Hot Observables, we can miss messages if we subscribe too late. Here, we create the Hot Observable immediately, but only subscribe to it after 1 second (thus missing the first message).
+
+```csharp
+var scheduler = new TestScheduler();
+var source = scheduler.CreateHotObservable(
+    new Recorded<Notification<long>>(10000000, Notification.CreateOnNext(0L)),
+    new Recorded<Notification<long>>(20000000, Notification.CreateOnNext(1L)),
+    new Recorded<Notification<long>>(30000000, Notification.CreateOnNext(2L)),
+    new Recorded<Notification<long>>(40000000, Notification.CreateOnNext(3L)),
+    new Recorded<Notification<long>>(40000000, Notification.CreateOnCompleted<long>())
+    );
+
+var testObserver = scheduler.Start(
+    () => source,
+    0,
+    TimeSpan.FromSeconds(1).Ticks,
+    TimeSpan.FromSeconds(5).Ticks);
+
+Console.WriteLine("Time is {0} ticks", scheduler.Clock);
+Console.WriteLine("Received {0} notifications", testObserver.Messages.Count);
+
+foreach (Recorded<Notification<long>> message in testObserver.Messages)
+{
+    Console.WriteLine("  {0} @ {1}", message.Value, message.Time);
+}
+```
+
+Output:
+
+```
+Time is 50000000 ticks
+Received 4 notifications
+OnNext(1) @ 20000000
+OnNext(2) @ 30000000
+OnNext(3) @ 40000000
+OnCompleted() @ 40000000
+```
+
+### CreateObserver 
```

Comment:
Provide a code sample?

Reply:
I think that might be better done as part of a rethink, incorporating some practical examples from the repo itself. (This was a chapter where I didn't change very much in this pass.)

So I've set up https://github.com/dotnet/reactive/issues/2069